src/hotspot/share/runtime/mutexLocker.hpp

```diff
@@ -188,21 +191,39 @@ void assert_lock_strong(const Mutex* lock);
 class MutexLockerImpl: public StackObj {
  protected:
   Mutex* _mutex;
+  bool _prof;
+  elapsedTimer _before;
+  elapsedTimer _after;
```

Comment:
It would be a lot clearer if these were also called `_wait_time` and `_hold_time`! And add a descriptive comment.

Reply:
I've changed them to `_wait_time` and `_hold_time` and added some comments.